akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala

```diff
@@ -97,7 +97,7 @@ private[parser] trait CommonRules { this: Parser with StringBuilding ⇒
 
   // def `rfc850-date` = rule { `day-name-l` ~ ", " ~ date2 ~ ' ' ~ `time-of-day` ~ " GMT" }
 
-  def date2 = rule { day ~ '-' ~ month ~ '-' ~ digit2 }
+  def date2 = rule { day ~ '-' ~ month ~ '-' ~ digit2 ~> { (year) => push(rfc6265Year(year))} }
```

Comment:
How about we simplify this to:

```scala
 ... ~> (y => if (y <= 69) y + 2000 else y + 1900) // https://tools.ietf.org/html/rfc6265#section-5.1.1
```

Saves us a few lines of code and is arguably even clearer.